Automattic/jetpack
# Related posts: append original user agent to API requests

#### Changes proposed in this Pull Request:


We want to be able to optimize responses for related posts
requested by crawlers and bots. Currently, we can't discern
those in backend from regular users. To allow that, I'm
proposing adding the original user agent string to the one
we're sending to API.

Additionally, I've also added avoiding caching Conflict
responses, even for those 60 seconds that we normally cache
empty responses for.

See also: D86909-code

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?

Looking at the https://jetpack.com/support/privacy/, we already mention user agent as one of the things that can be sent with each Automattic's system usage, so there's no need for update.




#### Testing instructions:






* Technically, there are no user-facing changes, so just visit a page with enabled related posts feature, and make sure it works as before.
* There will be some separate testing for D86909-code
